ansible/group_vars/searchengine-hosts.yml

```diff
@@ -14,3 +14,5 @@ cache_rows: 100000
 searchengine_secret_key: "fagfdssf3fgdnvhg56ghhgfhgfgh45f"
 searchengineurlprefix: "searchengine"
 IDR_TEST_FILE_URL: "https://raw.githubusercontent.com/IDR/idr.openmicroscopy.org/master/_data/studies.tsv"
+noelasticsearchnodes: 3
+elasticsearch_backup_folder: "path/to/elasticsearch/backup/folder"
```

Comment:
Should this be set to `/data/searchengine/backup` or similar?

Reply:
I think the backup should not be saved to the same data drive. Now I am using my home folder for testing but I am not sure if we can mount a drive (10-15 GB) to save the backup.

Reply:
Mounting a new hard disk is doable but before doing so, it would be good to agree on the expectations for the backup functionality. The PR does not include any description and I don't know if there is a relevant issue that I should read first.

Is this for recovery purposes in the case of data loss? Tailored to the multi-node scenario? Or to facilitate our rolling deployment strategy?

Reply:
Sorry, my fault; the backup can be used when the index data is corrupted for any reason. Something like updating the searchengine corrupted the elasticsearch data folder as I have mentioned today.
In such a case and re-indexing the data is not required,  we can use the backup copy to restore the data, if I remember correctly it takes about 20 minutes to restore the data from the backup copy, I will test it again to have an exact time.
The searchengine will back up the data automatically after indexing or -re-indexing the data. 
There is a method inside manage.py to backup the data (`backup_elasticsearch_data`) and another one to restore the data (`restore_elasticsearch_data`).
It can be used with one node or multiple nodes.

Reply:
The current production searchengine looks as follows

```
[sbesson@prod110-searchengine ~]$ df -h
Filesystem      Size  Used Avail Use% Mounted on
devtmpfs         32G     0   32G   0% /dev
tmpfs            32G     0   32G   0% /dev/shm
tmpfs            32G  700K   32G   1% /run
tmpfs            32G     0   32G   0% /sys/fs/cgroup
/dev/vda1       100G  7.3G   93G   8% /
/dev/vdb         75G  8.7G   67G  12% /data
tmpfs           6.3G     0  6.3G   0% /run/user/11615
```

My primary concern with creating yet-another-volume is that we have already allocated 75G to the searchengine volume (using SSD) but we are are only using a fraction of it.
If saving the backup on the same volume as the data is a concern, could we simply use the root partition which has lots of free space available?

Reply:
With an Elasticsearch cluster with three nodes, we will use more space (about 36%) of the data drive for the current IDR data.
OK, I think that is fine to create a backup folder on the root partition and use it for the backup.

Reply:
Looking at the testing installation
```
jburel@test112-searchengine ~]$ df -h
Filesystem      Size  Used Avail Use% Mounted on
devtmpfs         32G     0   32G   0% /dev
tmpfs            32G     0   32G   0% /dev/shm
tmpfs            32G  920K   32G   1% /run
tmpfs            32G     0   32G   0% /sys/fs/cgroup
/dev/vda1       100G   94G  6.1G  94% /
/dev/vdb         75G   26G   50G  34% /data
tmpfs           6.3G     0  6.3G   0% /run/user/15600
tmpfs           6.3G     0  6.3G   0% /run/user/5070
```
There is a massive jump in terms of usage.

Reply:
while testing I was keeping some copies in my home folder, I have deleted them now.
```
[kmohamed001@test112-searchengine temp]$ df
Filesystem     1K-blocks     Used Available Use% Mounted on
devtmpfs        32877712        0  32877712   0% /dev
tmpfs           32903212        0  32903212   0% /dev/shm
tmpfs           32903212      920  32902292   1% /run
tmpfs           32903212        0  32903212   0% /sys/fs/cgroup
/dev/vda1      104846316 64049832  40796484  62% /
/dev/vdb        78604800 26219688  52385112  34% /data
tmpfs            6580644        0   6580644   0% /run/user/15600
tmpfs            6580644        0   6580644   0% /run/user/5070
```

Reply:
I'd argue that even an increase from 8% (on `prod111`) to 62% brings us closer to the standard 80/90 threshold above which action should be taken.
Is it possible part of this comes from intermediate Docker images and containers? If so, it might be worth running `docker system prune -a` to see if that brings the usage down further

Reply:
After pruning
``` 
df -h
Filesystem      Size  Used Avail Use% Mounted on
devtmpfs         32G     0   32G   0% /dev
tmpfs            32G     0   32G   0% /dev/shm
tmpfs            32G 1000K   32G   1% /run
tmpfs            32G     0   32G   0% /sys/fs/cgroup
/dev/vda1       100G   17G   84G  17% /
/dev/vdb         75G   26G   50G  34% /data
tmpfs           6.3G     0  6.3G   0% /run/user/15600
tmpfs           6.3G     0  6.3G   0% /run/user/5070
```
the size increase is due to the backup. This does not need to be stored alongside the "prod" data.